apps/modal/src/functions/test_whisper.py

```
@@ -0,0 +1,63 @@
+        # But 'download_root' is the key. It checks if model exists there.
+        model = whisper.load_model("large-v3", device=device, download_root=model_path)
+        print("Model loaded successfully!")
+    except Exception as e:
+        print(f"❌ Failed to load model: {e}")
+        return
+
+    # Download a sample audio file (short extract)
+    print("Downloading sample audio for test...")
+    sample_url = "https://huggingface.co/datasets/Xenova/transformers.js-docs/resolve/main/jfk.wav"
+    sample_path = "jfk.wav"
+    urllib.request.urlretrieve(sample_url, sample_path)
```

Ruff (0.14.14)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)